dask/array/core.py

```diff
@@ -3097,7 +3097,9 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None, previous_chunks
     if shape is not None:
         chunks = tuple(c if c not in {None, -1} else s for c, s in zip(chunks, shape))
 
+    nonans = None
     if chunks and shape is not None:
+        nonans = all(isinstance(c, int) for c in chunks)
```

Comment:
Is there a reason to  use an `int` check instead of `not math.isnan(c)`? The `nonans` name seems to be slightly different than the code that's in this line

Reply:
The `c` in this case can be tuples - we are checking specifically for individual ints. 

Also, it's a bit faster eve
```
In [9]: l = [1] * 100

In [10]: %timeit all(isinstance(x, int) for x in l)
3.41 µs ± 14.2 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [11]: %timeit all(not math.isnan(x) for x in l)
5.55 µs ± 8.38 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```